packages/core/src/builder.class.ts

```diff
@@ -1939,7 +1939,11 @@ export class Builder {
 
           // The whole response has been received. Print out the result.
           resp.on('end', () => {
-            resolve(JSON.parse(data));
+            try {
+              resolve(JSON.parse(data));
+            } catch (err) {
+              reject(err);
```

Comment:
@steve8708 in the past based on the bug @tonytamps reported #476  we used to get data as an HTML page if the machine that's running the query flagged as a bot. This PR seems to be fixing the crash, but we would still reject here which is why on our code what happens is:
```
builder
 .get(model, builderOptions)
 .promise()
 .then((res) => {
   // This code is never reached
  })
  .catch((ex) => {
    // This is where we log the unexpected token
   })
```
and the error we get in the rejection fits a JSON.parse exception `Unexpected token < in JSON at position 0`.

this seems to be the root cause of the outage in the last week of October. What can we change on our end to use the SDK in such a way that we won't get flagged as a bot by any of your services?

Reply:
ah yes - in the cases where we failed to fetch data, I believe the above behavior is correct right? `.catch` will run forwarding whatever unexpected error occurred, as opposed to `.then` as it was not a success

re: bot detection, we actually found out what the issue was and fortunately it was not bot detection. Fastly had a POP node with a poisoned cache that was not being purged correctly. the reason that switching off of the SDK and to the REST APIs circumvented this is that the API parameters sent manually are slightly different than what the SDK sends so we didn't have any more usage of the poisoned caches (as the full URL is the cache key)

will have more info for you on this later